packages/next/next-server/lib/router/router.ts

```diff
@@ -216,13 +216,17 @@ export default class Router implements BaseRouter {
     this.isFallback = isFallback
 
     if (typeof window !== 'undefined') {
-      // in order for `e.state` to work on the `onpopstate` event
-      // we have to register the initial route upon initialization
-      this.changeState(
-        'replaceState',
-        formatWithValidation({ pathname, query }),
-        as
-      )
+      // make sure "as" doesn't start with double slashes or else it can
+      // throw an error as it's considered invalid
+      if (as.substr(0, 2) !== '//') {
```

Comment:
If this is skipped doesn't that mean that it'll break back navigation

Reply:
It looks like if `next/link` is used it will throw another error from `native-url` failing to handle parsing `//` https://github.com/zeit/next.js/blob/b1aa594a80e3af13ad22f9c97eab346f36242222/packages/next/client/link.tsx#L16

Although if `router.push` is used navigating works fine. I added a test case for navigating via `router.push`, not sure how we want to handle this since `next/link` would throw 🤔looks like it may need to be fixed in `native-url` since the normal `url` module handles this and returns

```
> url.parse('//')
Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: '//',
  path: '//',
  href: '//'
}
```